spec/system/candidate_interface/candidate_personal_details_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+RSpec.feature 'Entering their personal details' do
+  scenario 'Logged in candidate with no personal details' do
```

Comment:
On GOV.UK we've found that the feature tests stay more maintainable if we limit ourselves to 1 scenario per file.

For a random example with multiple scenarios: 

https://github.com/alphagov/content-tagger/blob/master/spec/features/project_spec.rb

Versus a spec with a single scenario:

https://github.com/alphagov/content-publisher/blob/master/spec/features/scheduling/propose_publish_time_spec.rb

In this case, I think you can merge the scenarios where the user completes the entire journey.

Reply:
I think this is a sound suggestion, whatever we can do to avoid having very long and cumbersome specs is good in my book.

In the `content-publisher` ones they do have some files with 2 scenarios: https://github.com/alphagov/content-publisher/blob/master/spec/features/scheduling/propose_publish_time_requirements_spec.rb

I think it makes sense to iterate on this over the next couple of PRs as we flesh out the user journey; this could be broken down e.g. a separate one for `editing`. 👍 

So I propose we keep this currently but look to change it in the follow-up PRs.

Reply:
Ooooo @tijmenb, I like the single scenario spec structure as it means we don't have a long list of given, when, thens after the scenarios.

How do you imagine our spec structure in this scenario?

Reply:
@chubberlisk if I'm not mistaken this is the flow:

```rb
scenario 'Candidate fills in personal details' do
  given_i_am_signed_in
  and_i_visit_the_site
  when_i_click_on_personal_details
  then_i_can_edit_my_personal_details

  when_i_fill_in_the_form
  then_i_can_check_my_answers

  when_i_click_on_back_to_application
  then_i_can_see_my_application
end
```

There's an important upside to having 1 scenario per file is that you can read it in-order both by reading the method calls (`given_i_am_signed_in`) and by the reading the method definitions top-to-bottom. The latter you can't do if the method defs are mixed.

Reply:
That's a good point @tijmenb!

Will do as @tvararu suggested though and do this in follow up PRs as we flesh this section out. Thinking we have ones for organising our specs folder and file structure, converting them to the given, when, then structure and merging scenarios where necessary?